examples/bond/assets/issuer-lsig.py

```diff
@@ -0,0 +1,67 @@
+from pyteal import *
+
+def issuer_lsig():
+    # check no rekeying, close remainder to, asset close to
+    common_fields = And(
+        Gtxn[0].rekey_to() == Global.zero_address(),
+        Gtxn[1].rekey_to() == Global.zero_address(),
+        Gtxn[2].rekey_to() == Global.zero_address(),
+        Gtxn[0].close_remainder_to() == Global.zero_address(),
+        Gtxn[1].close_remainder_to() == Global.zero_address(),
+        Gtxn[2].close_remainder_to() == Global.zero_address(),
+        Gtxn[1].asset_close_to() == Global.zero_address(),
+    )
+
+    # verify that buyer deposits required algos 
+    verify_tx = And(
+        Gtxn[0].type_enum() == TxnType.Payment,
+        Gtxn[2].application_id() == Tmpl.Int("TMPL_APPLICATION_ID"),
+        Gtxn[2].application_args[0] == Bytes("buy"),
+        common_fields
+    )
+
+    # burn tx
+    burn_tx = And(
+        Gtxn[2].type_enum() == TxnType.AssetTransfer,
+        Gtxn[1].type_enum() == TxnType.AssetTransfer,
+        Gtxn[0].application_id() == Tmpl.Int("TMPL_APPLICATION_ID")
```

Comment:
Should we assert first that `Gtxn[0].type_enum() == TxnType.NoOp` ?

Reply:
yes

Reply:
added and ordered